src/Compilers/CSharp/Test/Symbol/Symbols/CovariantReturnTests.cs

```diff
@@ -27,6 +86,85 @@ private static void VerifyNoOverride(CSharpCompilation comp, string methodName)
             Assert.Null(overridden);
         }
 
+        /// <summary>
+        /// Verify that all assignments in the compilation's source have the same type and converted type.
+        /// </summary>
+        private static void VerifyAssignments(CSharpCompilation comp)
+        {
+            foreach (var tree in comp.SyntaxTrees)
+            {
+                var model = comp.GetSemanticModel(tree);
+                foreach (var assignment in tree.GetRoot().DescendantNodes().OfType<AssignmentExpressionSyntax>())
+                {
+                    var right = assignment.Right;
+                    var typeInfo = model.GetTypeInfo(right);
+                    Assert.True(typeInfo.Type.GetSymbol<TypeSymbol>().Equals(typeInfo.ConvertedType.GetSymbol<TypeSymbol>(), TypeCompareKind.AllIgnoreOptions));
```

Comment:
>TypeCompareKind.AllIgnoreOptions [](start = 125, length = 32)

We could add a singleton for this option into Microsoft.CodeAnalysis.SymbolEqualityComparer and use it to compare ITypeSymbols without having to dig to the internal symbols.

Reply:
This test is replaced by a type equality test on the public symbol model in [https://github.com/dotnet/roslyn/pull/44025](https://github.com/dotnet/roslyn/pull/44025)

---
In reply to: [421865677](https://github.com/dotnet/roslyn/pull/43797#discussion_r421865677) [](ancestors = 421865677)